magento/pwa-studio
# FilterModal talon

## Description

Create Peregrine talons for the `FilterModal` component. As part of this work, refactor the component to use local state and hooks throughout, and improve its performance.

## Related Issue
Closes #1930 .

## Acceptance 
@sirugh 

### Verification Stakeholders
@dpatil-magento 

### Specification

### Verification Steps
1. Navigate to a category page
2. Click `Filter` to open the filter modal (drawer)
3. Select some filters
4. Click `Apply Filters` to close the drawer and update the URL
5. Refresh the page
6. Reopen the filter modal and verify that the state remains correct
7. Click `Reset Filters` to clear all of the filters
8. Click the mask or the `X` to close the drawer and update the URL
9. Repeat this process on the search page

## Screenshots / Screen Captures (if appropriate)

## Checklist
* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
